src/engraving/rendering/dev/tlayout.cpp

```diff
@@ -5911,7 +5911,15 @@ void TLayout::layoutTextLineBaseSegment(TextLineBaseSegment* item, LayoutContext
         double gapBetweenTextAndLine = _spatium * tl->gapBetweenTextAndLine().val();
         if ((item->isSingleBeginType() && (tl->beginTextPlace() == TextPlace::LEFT || tl->beginTextPlace() == TextPlace::AUTO))
             || (!item->isSingleBeginType() && (tl->continueTextPlace() == TextPlace::LEFT || tl->continueTextPlace() == TextPlace::AUTO))) {
-            l = item->text()->pos().x() + item->text()->ldata()->bbox().width() + gapBetweenTextAndLine;
+            l = item->text()->pos().x() + gapBetweenTextAndLine;
+            switch (item->text()->align().horizontal) {
+            case AlignH::LEFT:
+                l += item->text()->ldata()->bbox().width();
+                break;
+            case AlignH::HCENTER:
+                l += item->text()->ldata()->bbox().width() / 2;
+                break;
+            }
```

Comment:
I believe a default case is needed to avoid compiler warnings reg. unhandled cases?

Reply:
Also: where is `AlignH::Right handled?

Reply:
> I believe a default case is needed to avoid compiler warnings reg. unhandled cases?

A few lines further up in this layout method, there is a switch with also no default case

Reply:
> Also: where is `AlignH::Right handled?

`AlignH::Right` does not need to be offset by the text width, since the text ends before point 0

Reply:
> I believe a default case is needed to avoid compiler warnings reg. unhandled cases?

Right, but that handles all possible cases

Reply:
> > I believe a default case is needed to avoid compiler warnings reg. unhandled cases?
> 
> A few lines further up in this layout method, there is a switch with also no default case

Right, but that handles all possible cases

Reply:
Simply add `default: break;`, to tell the compiler explicitly that you didn't forget the other two cases

Reply:
Exactly my point. From the CI build logs:
```
2024-04-28T07:30:02.2925013Z ../src/engraving/rendering/dev/tlayout.cpp: In static member function ‘static void mu::engraving::rendering::dev::TLayout::layoutTextLineBaseSegment(mu::engraving::TextLineBaseSegment*, mu::engraving::rendering::dev::LayoutContext&)’:
2024-04-28T07:30:02.2993137Z ../src/engraving/rendering/dev/tlayout.cpp:5924:20: warning: enumeration value ‘RIGHT’ not handled in switch [-Wswitch]
2024-04-28T07:30:02.2994430Z  5924 |             switch (item->text()->align().horizontal) {
2024-04-28T07:30:02.2995009Z       |                    ^
2024-04-28T07:30:02.2996367Z ../src/engraving/rendering/dev/tlayout.cpp:5970:16: warning: enumeration value ‘LEFT’ not handled in switch [-Wswitch]
2024-04-28T07:30:02.2997694Z  5970 |         switch (item->endText()->align().horizontal) {
2024-04-28T07:30:02.3109675Z       |                ^
```